### DISCUSSION TITLE
ChangeLog

### DISCUSSION THREAD

**Discussion by m-holger** [qpdf/qpdf-dev#7] [General]
> The ChangeLog format conforms to the gnu format, including the particular line break and indentation style, so let's not reformat ChangeLog. My goal for ChangeLog is that it's a level of granularity between the release notes and the git history. It gives someone a chance to scan the list of changes from one release to another and see what might be relevant. Also, ChangeLogs are chronological, while the release notes are organized by category. I try to do ChangeLog as I go. When working on features, I try to do release notes as I go as well, but I'm also likely to build the release notes by looking at the ChangeLog and diffing public headers.

I wasn't aware of the gnu format. The only info I was able to find was https://www.gnu.org/prep/standards/standards.html#Change-Logs, which is silent on some of the detail. Looking at the ChangeLog itself there is a mixture of indentation using spaces and indentation using a single tab (equivalent to 8 spaces). I am not sure what is intended.

**Comment by jberkenbilt** [comment]
Looks like tabs are intended. I use emacs ChangeLog mode to edit my ChangeLog files, and it keeps everything conformant. I did a direct push to main with a commit that fixes whitespace in ChangeLog and updates .git-blame-ignore-revs.

**Comment by m-holger** [comment]
Do you have a way to edit the change log using CLion? I can't work out how to change the indentation style to use tabs for a single file and had to revert back to using my trusted old Geany editor for the change log.

**Reply by m-holger** [reply]
This is not a problem, just a question about CLion

**Reply by jberkenbilt** [reply]
Okay, I don't really know. I only use emacs for ChangeLogs. But as I said, I'm not married to ChangeLogs. It's cumbersome to have three different ways to track changes (git history, ChangeLog, release notes). ChangeLog comes from a time when people didn't have access to the version control system and people were using tools like RCS that are single-file based. I really think the idea of ChangeLog is obsolete and we should probably just rethink how to do release notes. But for now, if you have a solution to maintain the format consistency, use that until this bubbles up on one or the other of our to-do lists.

**Comment by jberkenbilt** [comment]
Whatever you're using to edit ChangeLog files is leaving trailing whitespace on some lines. Those are visible in emacs. If this is too hard to fix, don't worry about it.

**Reply by m-holger** [reply]
It requires a highly technical fix (i.e. tick the 'newline strips trailing spaces' box)

**Reply by jberkenbilt** [reply]
Okay, if you can work that into your busy schedule, please do. It would probably take me three months, but you are usually more responsive.
